tests/protocol/mocks.py

```diff
@@ -101,15 +97,17 @@ def http_request(method, url, data, **kwargs):
                 return return_value
 
         # if the request was not handled try to use the mock wireserver data
-        if _USE_MOCK_WIRE_DATA_RE.match(url) is not None:
+        try:
             if method == 'GET':
                 return protocol.mock_wire_data.mock_http_get(url, **kwargs)
             if method == 'POST':
                 return protocol.mock_wire_data.mock_http_post(url, data, **kwargs)
             if method == 'PUT':
                 return protocol.mock_wire_data.mock_http_put(url, data, **kwargs)
+        except NotImplementedError:
```

Comment:
Is NotImplementedError a temporary error we raise for development purpose ? Just want to make sure why we are not doing anything with it after catching.

Reply:
Yes, good comment. I am not totally happy with the choice of NotImplemented, since that exception sort of gives the impression that this is temporary, as you point out. I used it as meaning that support for the specific "url" given as parameter is not supported by the mock_http_* functions (you can see when this exception is raised in these functions). Initially I considered a TypeError. but that is currently used to indicated a bad request (e.g incorrect headers). In the cases where NotImplemented is raised for a specific request, it is totally possible to add support for that request in the mocks, even if right now they are not supported, so I went for that error.

As far as why we are not doing anything in the except clause, that is explained in a comment a couple of lines below: if the mock_http_* functions don't handle the request we need to continue and fail the request or pass it to the original function we are mocking, depending on the flag passed as argument to this function.

Reply:
Makes sense. Thanks for explaining.